src/Common/TLDListsHolder.cpp

```diff
@@ -15,20 +15,32 @@ namespace ErrorCodes
     extern const int LOGICAL_ERROR;
 }
 
+constexpr size_t StringHashTablePadRequirement = 8;
+
 /// TLDList
 TLDList::TLDList(size_t size)
     : tld_container(size)
-    , pool(std::make_unique<Arena>(10 << 20))
-{}
-bool TLDList::insert(StringRef host)
+    , memory_pool(std::make_unique<Arena>())
+{
+    /// StringHashTable requires padded to 8 bytes key,
+    /// and Arena (memory_pool here) does satisfies this,
+    /// since it has padding with 15 bytes at the right.
+    ///
+    /// However, StringHashTable may reference -1 byte of the key,
```

Comment:
Here, since `size` can be less then 8

https://github.com/ClickHouse/ClickHouse/blob/d0f5dcad25785fc278587e6a5a883ecbaea5b545/src/Common/HashTable/StringHashTable.h#L286-L287